src/sql-parser/src/parser.rs

```diff
@@ -1721,6 +1724,32 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_create_role(&mut self) -> Result<Statement<Raw>, ParserError> {
+        let is_user = match self.expect_one_of_keywords(&[ROLE, USER])? {
+            ROLE => false,
+            USER => true,
+            _ => unreachable!(),
+        };
+        let name = self.parse_identifier()?;
+        let _ = self.parse_keyword(WITH);
+        let mut options = vec![];
+        loop {
```

Comment:
idk if this matters but pg seemingly does type-like parsing for these options and only parses those with > 1 token:

```
sean=#  CREATE ROLE usr WITH badopt ;
ERROR:  unrecognized role option "badopt"

sean=#  CREATE ROLE usr WITH CONNECTION;
ERROR:  syntax error at or near ";"
LINE 1: CREATE ROLE usr WITH CONNECTION;
```

Reply:
Sorry––wrote that right before bed. Not very clear!

If the text following `WITH` is not a token known to be part of a multi-token option, PG parses it as an option––the content of the options are then validated later. The consequence of this is that `SUPERUSER` et. al. are not tokens within PG, and checking for valid options isn't done in parsing, but instead in planning.

Don't think it matters much––I just know how much you love minimizing the number of unnecessary tokens we have strewn about.

Reply:
Ohh, I see! Good point. I'm not terribly worried about these extra keywords (nonrestricted keywords in our SQL parser are essentially zero cost, which is not the case in PostgreSQL) but definitely possible that we'll need to clean this up in the future.